server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -616,13 +627,22 @@ public boolean hasExplicitRestApiVersion() {
         return restApiVersion.isPresent();
     }
 
-    public void markPathRestricted(String restriction) {
-        if (params.containsKey(PATH_RESTRICTED)) {
-            throw new IllegalArgumentException("The parameter [" + PATH_RESTRICTED + "] is already defined.");
+    public void setUseServerlessPartialApiRestrictions() {
+        if (params.containsKey(USE_SERVERLESS_PARTIAL_API_RESTRICTIONS)) {
+            throw new IllegalArgumentException("The parameter [" + USE_SERVERLESS_PARTIAL_API_RESTRICTIONS + "] is already defined.");
         }
-        params.put(PATH_RESTRICTED, restriction);
+        params.put(USE_SERVERLESS_PARTIAL_API_RESTRICTIONS, "true");
```

Comment:
This is a "functional" change: it specializes the flag to serverless mode restrictions (as opposed to providing abstract support for more general restrictions). Given that there are no concrete plans for other types of restrictions I think specializing this for serverless mode is worthwhile since it makes things easier to understand and easier to reason about.